### PR TITLE
Fix revision REST API related issues

### PR DESCRIPTION
**Pull request by CrowleyRajapakse**
## Implementation

Fixing REST API docs
Adding apim:api_create scope to revision resources

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml [hunk 1]
```diff
@@ -8641,9 +8666,12 @@ components:
           type: string
           example: mg.wso2.com
         displayOnDevportal:
+          maxLength: 255
+          minLength: 1
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml:8670]: is this correct for a boolean?

Comment (CrowleyRajapakse) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml]: removed them and fixed

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml [hunk 2]
```diff
@@ -1555,7 +1559,7 @@ paths:
     get:
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml:1559]: Here the resource path should be `GET /apis/{apiId}/deployed-revisions`

Comment (CrowleyRajapakse) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml:1558]: fixed and resolved

**Review by malinthaprasan — APPROVED** [review]

**Pull request merged**